### ISSUE TITLE
A10x not working, hangs on patch stage

### ISSUE THREAD

**Issue by verygenericname** [0x7ff/gaster#54]
![image](https://user-images.githubusercontent.com/87825638/235267000-daac4d8c-a7f3-47b0-bd70-910ae9b724b0.png)

**Comment by rastiqdev** [comment]
if it stops at patch i think you entered pwndfu successfully but i'm not sure

**Comment by Aditya20110** [comment]
Means the device is not in either actual dfu mode or there is some issue with usb. Are u using usb A right? And it happened to me also one time when i was tether downgrading my ios it downgraded but after boot the same prob. Was happening with me,better try to use palera1n dfu helper.

**Comment by verygenericname** [comment]
its happening on multiple computers, i've used multiple usb a cables, it just disconnects and isn't recognized anymore

**Comment by verygenericname** [comment]
Also I have used dfu helper

**Comment by verygenericname** [comment]
A older version of gaster fixed this, too

**Comment by Aditya20110** [comment]
Did u restarted usbmuxd?

**Comment by verygenericname** [comment]
usbmuxd shouldn't do anything with dfu, and it happens on macOS and Linux

**Comment by Aditya20110** [comment]
Anyways,i wanted to talk to you. Can u tell ur discord?

**Comment by verygenericname** [comment]
Nathan#5457

**Comment by Aditya20110** [comment]
Ahh bro. U dont accept friend requests. Add me pls Aditya2011#2218

**Comment by alfiecg24** [comment]
Having done some testing, it seems that A10X support was broken in commit 7fffffffabbd6f5737377d8eebb924f4bc67640e, AKA "A7 Fix". I may try looking into it to see what the problematic change was exactly.

**Comment by alfiecg24** [comment]
@0x7ff @verygenericname it seems that commenting out [these two lines](https://github.com/0x7ff/gaster/tree/main/gaster.c#L1218-L1219) will fix the issue on A10X - however, as this is part of the A7 fix, I am not sure if this will break support for other devices. Therefore, I think that it will require additional testing before implementing into the main branch.

**Comment by alfiecg24** [comment]
Those two extra control transfers are necessary on A9X, as I have just tested with and without, yet still cause the exploit to fail on A10X.

**Comment by verygenericname** [comment]
My fork fixed it
i just added a a10x check and make those 2 lines do nothing if its a10x

**Comment by alfiecg24** [comment]
Yeah I’ve been using the fix in my checkm8 project and it works fine, just might be worth checking if A10/A11 also have the same issue?

**Comment by verygenericname** [comment]
They don't

**Comment by alfiecg24** [comment]
Interesting, thanks!

**Comment by ghost** [comment]
ive got this issue on my ipad 7th gen A10, any one know how to fix it?

**Comment by ghost** [comment]
i fixed the issue by going into REAL DFU. i wasn't in real dfu i was in LLB DFU (or smth like that)
issue fixed tho
